packages/kbn-openapi-generator/src/template_service/templates/zod_operation_schema.handlebars

```diff
@@ -36,8 +36,21 @@ export const {{@key}} = {{> zod_schema_item}};
 {{/if}}
 {{#if enum}}
 {{#unless (isSingle enum)}}
-export type {{@key}}Enum = typeof {{@key}}.enum;
-export const {{@key}}Enum = {{@key}}.enum;
+  {{#if default }}
+  {{!
+      `zod_schema_item` generated enum with a default value via `.default()`. It has type `ZodDefault`.
+      At the same time `ZodDefault` doesn't have `enum` member causing type checking error.
+      As per https://github.com/colinhacks/zod/issues/290 to fix it original enum should be used.
+      The solution below generates enum without a default value making the code below passing the
+      type check.
+  }}
+  const {{@key}}NonDefault = {{> zod_enum}}
+  export type {{@key}}Enum = typeof {{@key}}NonDefault.enum;
+  export const {{@key}}Enum = {{@key}}NonDefault.enum;
+  {{else}}
+  export type {{@key}}Enum = typeof {{@key}}.enum;
+  export const {{@key}}Enum = {{@key}}.enum;
```

Comment:
What is the use case for this `NonDefault` enum? I'm trying to better understand how it should be used.

I added a `Test` enum with a default value. The generator returned an expected error, stating that there should not be default values, and suggested appending `Default` to the enum as an alternative. Pretty good DX here 👍

Then, I renamed the enum to `DefaultTest` and ran the generator script. This time, there were no errors, but the produced code looked like this:

```ts
export type DefaultTest = z.infer<typeof DefaultTest>;
export const DefaultTest = z.enum(['one', 'two', 'three']).default('one');
const DefaultTestNonDefault = z.enum(['one', 'two', 'three']);
export type DefaultTestEnum = typeof DefaultTestNonDefault.enum;
export const DefaultTestEnum = DefaultTestNonDefault.enum;
```

If the idea is to produce two structures - one with the default value and another without - I think the code should look like this:

```ts
// First, define the enum type without the default value
const Test = z.enum(['one', 'two', 'three']);
export type TestEnum = typeof Test.enum;
export const TestEnum = Test.enum;

// Then reuse the enum to define the type with the default value
export type DefaultTest = z.infer<typeof DefaultTest>;
export const DefaultTest = Test.default('one');
```